File: 7-kyu-kata/reverse-factorials/python/reverse_factorials_test.py
# 7 kyu - Reverse Factorials  [ ID: 58067088c27998b119000451  (reverse-factorials) ]
# URL: https://www.codewars.com/kata/58067088c27998b119000451
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
test.describe("Static Cases")
test.assert_equals(reverse_factorial(120), "5!")
test.assert_equals(reverse_factorial(3628800), "10!")
test.assert_equals(reverse_factorial(150), "None")

test.describe("Random Tests")
from random import shuffle, randint

my_factorials_341 = [
    1,
    2,
    6,
    24,
    120,
    720,
    5040,
    40320,
    362880,
    3628800,
    39916800,
    479001600,
    6227020800,
    87178291200,
    1307674368000,
]

test_cases = my_factorials_341 * 2 + [randint(3, 1307674368000) for _ in range(50)]
shuffle(test_cases)

for num in test_cases:
    if num in my_factorials_341:
        test_exp = str(my_factorials_341.index(num) + 1) + "!"
    else:
        test_exp = "None"
    test.it("Testing for num = %s, expecting = '%s'" % (num, test_exp))
    test.assert_equals(reverse_factorial(num), test_exp)

File: 7_Kyu_Kata/reverse-factorials/JavaScript/reverseFactorials.Test.js
// [object Object] - Reverse Factorials  [ ID: 58067088c27998b119000451  (reverse-factorials) ]
// URL: https://www.codewars.com/kata/58067088c27998b119000451
// Category: REFERENCE  |  Tags: FUNDAMENTALS
// *****************************************************************************
describe("Tests", () => {
  it("test", () => {
Test.assertEquals(reverseFactorial(120), '5!');
Test.assertEquals(reverseFactorial(3628800), '10!');
Test.assertEquals(reverseFactorial(150), 'None');

function reverseFactorialCheck(num) {
  let x = 1;
  while (num > 1 && Math.round(num) === num) {
    x += 1;
    num /= x;
  }
  if (num === 1) {
    return x + '!';
  } else {
    return 'None'
  }
}
for(let i = 0; i < 100; i++){
  let random1 = Math.round(Math.random() * 10);  
  let random2 = Math.round(Math.random() * 10);  
  Test.assertEquals(reverseFactorial(random1, random2), reverseFactorialCheck(random1, random2))
}
  });
});

